src/binary-reader.cc

```diff
@@ -1649,10 +1652,14 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
         ++i;
         break;
       default:
-        /* unknown subsection, skip rest of section */
-        ctx->offset = ctx->read_end;
+        /* unknown subsection, skip it */
```

Comment:
The existing code allows the consumer to check the sections are in the right order (it doesn't verify that the function/local indices are specified in order either, but again the consumer can check that).

Now that I think about it, I feel that stricter is better, and we shouldn't go out of our way to be lenient.

I must confess I've resorted a few times to cutting out the name section of wasm modules in a text editor, so a switch to ignore (some) custom sections would make sense.